### DISCUSSION TITLE
A simple, straightforward Monte-Carlo simulation in R

### DISCUSSION THREAD

**Discussion by thuyduyenph249** [HTAhackathon2024/.github#22] [Ideas]
### What is the problem we're trying to solve

An easy, straightforward Monte-Carlo simulation package for probabilistic analysis. For example, we have 3 input indicators that could be put together in one formula to calculate an outcome Y (for example: Y = X1 * X2 + X3). To do the PSA, we will simultaneously and randomly pick X1, X2, X3 from their distributions for 1,000 iterations to get 1,000 corresponding values for Y. The vector of 1,000 values for Y could be stored separately for further estimations/ algorithms if possible. 
I am not so sure if there is any existing package for this procedure but one easy way to do this is using VBA in Excel. It would be nice if we could do it in R where we could keep track of everything in one script without jumping back and forth several tabs. 

### How will we tackle the problem?

1.	Recheck if there is any package have already solved this task
2.	Outline a simple, straightforward Monte-Carlo simulation like the example above and test it out with some existing data. I plan for an simple command in this form:
mcs (Y = X1*X2+X3, dist(X1)= … , dist(x2) =…, dist(X3) =…, iterationnumber)
3.	Further improve the Monte-Carlo simulation (as from my humble knowledge, the example above only touches the surface of this technique)

**Comment by n8thangreen** [comment]
what about something like the [simsalapar](https://cran.r-project.org/web/packages/simsalapar/index.html) package?

**Comment by aponzo125** [comment]
I would be interested in this. I have mainly used Excel for modeling so would be really keen to work on monte carlo simulation in r
